### PR TITLE
Feature: CmdParse can construct lists by providing an 'append' style option

### PR DESCRIPTION
**Pull request by schwager-hsph**
Akin to the optparse 'append' action (https://docs.python.org/2/library/optparse.html#other-actions).

Providing multiple flags on the command line constructs a list.
For example, `doit -l foo -l bar` translates to `['foo','bar']` for the `-l` flag.

### CODE REVIEW

**Comment by schwager-hsph** [comment]
Sorry, didn't mean to add in the PythonAction commit.

File: tests/test_cmdparse.py [hunk 1]
```diff
@@ -153,26 +160,33 @@ def test_getOption(self, cmd):
...
+        assert ['foo', 'bar'] == params['list']
 
     def test_parseLongValues(self, cmd):
-        params, args = cmd.parse(['--rare-bool','--num','89', '--no-flag'])
+        params, args = cmd.parse(['--rare-bool','--num','89', '--no-flag',
+                                  '--list', 'flip', '--list', 'flop'])
         assert True == params['rare']
         assert False == params['flag']
         assert 89 == params['num']
+        assert ['foo', 'bar', 'flip', 'flop'] == params['list']
```
Comment (schettino72) [inline tests/test_cmdparse.py]: looks like a bug that `foo` and `bar`  show up here.

File: tests/test_cmdparse.py [hunk 2]
```diff
@@ -153,26 +160,33 @@ def test_getOption(self, cmd):
...
 
     def test_parsePositionalArgs(self, cmd):
-        params, args = cmd.parse(['-f','p1','p2','--sub-arg'])
```
Comment (schettino72) [inline tests/test_cmdparse.py]: why you removed this?

File: doit/cmdparse.py [hunk 3]
```diff
@@ -213,8 +213,8 @@ def parse(self, in_args):
 
         # parse options using getopt
         try:
-            opts, args = getopt.getopt(in_args, self.get_short(),
-                                       self.get_long())
+            opts, args = getopt.gnu_getopt(in_args, self.get_short(),
```
Comment (schettino72) [inline doit/cmdparse.py]: I guess this change will break other parts of the code.

File: tests/test_cmdparse.py (hunk 2 above)
Comment (schwager-hsph) [inline tests/test_cmdparse.py]: I changed this in a627702 to give `cmd.parse` a list of recognized options. `gnu_getopt` picks up on these later-placed options now.

File: tests/test_cmdparse.py (hunk 1 above)
Comment (schwager-hsph) [inline tests/test_cmdparse.py]: Fixed in cfd8b62 and 509cb4e. Thanks for the good catch!

**Comment by schwager-hsph** [comment]
Upon further review, I found that I didn't really need `gnu_getopt` in my code, either. I just needed to subclass `doit.cmd_run:Run` correctly. Sorry about that!

**Comment by schettino72** [comment]
please also add an entry on CHANGES, and docs ( I guess the arguments section should mention lists are also acceptable).

File: doit/cmdparse.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 """
 import getopt
 import six
-
```
Comment (schettino72) [inline doit/cmdparse.py]: I tend to use more blank lines than usual. IMO this increases readability... please dont remove my blank lines :D

Comment (schwager-hsph) [inline doit/cmdparse.py]: My mistake. I'll be more careful about blank lines :)

**Comment by schettino72** [comment]
merged. thanks!

**Pull request closed without merging**